src/data/finset.lean

```diff
@@ -2003,6 +2003,12 @@ begin
   exact nat.le_pred_of_lt h
 end
 
+lemma succ_bot {m n : ℕ} (h : m < n) : insert m (Ico (succ m) n) = Ico m n :=
```

Comment:
Also `m ≤ n`.

Reply:
And what is the difference with the next lemma `eq_cons`? (weird name, by the way...)

Reply:
There isn't a difference. I just hadn't seen that lemma, probably because it's name isn't very good.

Reply:
Also `m < n` is required, otherwise the right hand side is empty, and the left hand side is not. I checked this time.
`example : insert 4 (Ico (succ 4) 4) ≠ Ico 4 4 := dec_trivial`

Reply:
Ah yes, sorry, I was thinking of `Icc`. Thanks for checking!

Reply:
There are `eq_cons` lemmas in `multiset` and `list` as well if you feel like renaming those while you're at it.